src/sonic-yang-models/yang-models/sonic-lldp.yang

```diff
@@ -0,0 +1,164 @@
+                        exchanged. Default is 30 seconds";
+                }
+
+                leaf multiplier {
+                    type uint8 {
+                        range "1..10" {
+                            error-message "Invalid LLDP multiplier value.";
+                        }
+                    }
+                    default 4;
+                    description
+                        "This multiplier value is used to determine the timeout
+                        interval (i.e. hello-time x multiplier value) after
+                        which LLDP neighbor entry is deleted.";
+                }
+
+                leaf system_name {
+                    type string;
+                    description
+                        "System administratively assigned name";
+                }
+
+                leaf system_description {
+                    type string;
+                    description
+                        "System description";
+                }
+
+                leaf supp_mgmt_address_tlv {
+                    type boolean;
+                    description
+                        "Suppress sending of Management Address TLV in LLDP frames";
+                }
+
+                leaf supp_system_capabilities_tlv {
+                    type boolean;
+                    default false;
+                    description
+                        "Suppress sending of System Capabilities TLV in LLDP frames";
+                }
+
+                leaf enabled {
+                    type boolean;
+                    default true;
+                    description
+                        "Enable/Disable LLDP";
+                }
+
+                leaf mode {
+                    type enumeration {
+                        enum RECEIVE;
+                        enum TRANSMIT;
+                    }
+
+                    description
+                        "RX/TX mode for LLDP frames";
+                }
+
+                //uses lldp_mode_config;
+            }
+        }
+
+        container LLDP_PORT {
+            list LLDP_PORT_LIST {
+                key "ifname";
+
+                leaf ifname {
+                    type leafref {
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                    }
+                    description
+                        "Reference of port on which LLDP to be configured.";
+                }
+
+                leaf enabled {
+                    type boolean;
+                    default true;
+                    description
+                        "Enable/Disable LLDP";
+                }
+
+                leaf mode {
+                    type enumeration {
+                        enum RECEIVE;
+                        enum TRANSMIT;
+                    }
+
+                    description
+                        "RX/TX mode for LLDP frames";
+                }
+                //uses lldp_mode_config;
```

Comment:
We should use the grouping "lldp_mode_config", any reason for this comment?

Reply:
The compilation fails with the usage of grouping (not related to this PR) hence its commented out, will uncomment once that issue is fixed. This was discussed in the PR review and the relevant information was also shared to yang sub-group over email.

Reply:
save as above comment.